### DISCUSSION TITLE
When training a model in PyTorch using practical examples, I found that there are 9 fewer data in the train_batches. I want to know the position of the missing data in the original dataset.

### DISCUSSION THREAD

**Discussion by zhuyuhang4** [AdaptiveMotorControlLab/CEBRA#129] [Q&A] [answered]
![image](https://github.com/AdaptiveMotorControlLab/CEBRA/assets/144974755/9beabbcc-75a5-4a61-9760-1e8f5d5da0e7)

**Comment by stes** [comment, marked as answer]
Hi @zhuyuhang4 , this happens when you use a model with a receptive field greater than 1. In your example, the `offset10-model` uses a receptive field of 10, and the `neural_model.get_offset()` will tell you the number of samples that are removed from the `.left` and `.right` side of the data.

Does that clarify the question?

**Comment by zhuyuhang4** [comment]
Thank you！

**Comment by stes** [comment]
(The high level API also uses padding, so you might want to have a look at the `transform()` function if you are interested in how it is handled there)

**Comment by stes** [comment]
If this addresses your Q, please click on "Mark as answer" above :) Otherwise happy to discuss further!

**Comment by zhuyuhang4** [comment]
![image](https://github.com/AdaptiveMotorControlLab/CEBRA/assets/144974755/22cfe6d5-8806-4e4b-a077-f8891e1223ef)
Okay, is this the source code location?
